droidlet/lowlevel/hello_robot/remote/utils.py

```diff
@@ -94,44 +117,76 @@ def goto(robot, xyt_position=None, translation_threshold=0.1, dryrun=False, dept
         y = xyt_position[1]    # in meters
         rot = xyt_position[2]  # in radians
 
+        print("goto: ", xyt_position)
+
         if sqrt(x * x + y * y) < translation_threshold:
-            print("translation distance ", sqrt(x * x + y * y))
+            print("translation distance too little,"
+                  " rotating and exiting", sqrt(x * x + y * y))
             print(f'rotate by {xyt_position[2], rot}')
             if not dryrun:
                 robot.base.rotate_by(rot)
                 robot.push_command()
                 time.sleep(0.05)
             return True
 
+        # robot is at (0, 0) because we're using base-frame
+        # Now, calculate the angle between (x, y) and (0, 0)
+        # so, atan2(y - 0, x - 0)
         theta_1 = atan2(y, x)
+
+        # distance between (x, y) and (0, 0)
         dist = sqrt(x ** 2 + y ** 2)
 
+        # atan2(y, x) is the angle between the positive x axis and the ray to the point (x, y)
+        # so, it will be:
+        #  0    <= angle <= pi/2   :  if +x, +y
+        # +pi/2 <= angle <= pi     :  if -x, +y
+        # -pi   <= angle <= -pi/2  :  if -x, -y
+        # -pi/2 <= angle <= 0      :  if +x, -y
+
+        # if theta_1 is between 0 and pi / 2, then you will turn left by a bit and move forward
+        # if theta_1 is between -pi/2 and 0, then you will turn right by a bit and move forward
+
         if theta_1 > pi / 2:
```

Comment:
do you need a less than check here `theta_1 <= pi`?

Reply:
atan2 wont return anything greater than  `pi`, because it will then be in the quadrant `-pi <= angle <= -pi/2`

Here's an example:

```
In [25]: x, y = -1, 0; math.atan2(y, x)
Out[25]: 3.141592653589793

In [26]: x, y = -1, -0.001; math.atan2(y, x)
Out[26]: -3.1405926539231266
```